vscode/src/data/analyzerResults.ts

```diff
@@ -35,27 +35,38 @@ function getSeverityFromCategory(category: Category | undefined): vscode.Diagnos
 }
 
 export const processIncidents = (
-  ruleSets: Immutable<RuleSet[]>,
+  incidents: Immutable<EnhancedIncident[]>,
 ): ReadonlyArray<[vscode.Uri, vscode.Diagnostic[]]> =>
-  ruleSets
-    .flatMap((ruleSet): [string, Immutable<Violation>][] =>
-      Object.entries(ruleSet.violations ?? {}),
-    )
-    .flatMap(([code, violation]): [string, vscode.DiagnosticSeverity, Incident][] => {
-      const severity = getSeverityFromCategory(violation.category);
-      return violation.incidents.map((it) => [code, severity, it]);
-    })
-    .filter(([, , incident]) => incident.uri)
-    .map(([code, severity, incident]) => {
+  incidents
+    .filter((incident) => incident.uri)
+    .map((incident) => {
       const uri = vscode.Uri.parse(incident.uri);
       const line = (incident.lineNumber || 1) - 1;
-      const message = incident.message || "No message provided";
       const range = new vscode.Range(line, 0, line, 0);
 
-      const diagnostic = new vscode.Diagnostic(range, message, severity);
+      // Create a detailed message with all available context
+      const message = [
+        incident.message || "No message provided",
+        `\n\nContext:`,
+        `- Ruleset: ${incident.ruleset_name ?? "Unknown Ruleset"}`,
+        `- Ruleset Description: ${incident.ruleset_description ?? "No description available"}`,
+        `- Violation: ${incident.violation_name ?? "Unknown Violation"}`,
+        `- Violation Description: ${incident.violation_description ?? "No description available"}`,
+        `- Category: ${incident.violation_category ?? "Uncategorized"}`,
+        incident.violation_labels?.length
+          ? `- Labels: ${incident.violation_labels.join(", ")}`
+          : "",
+      ]
+        .filter(Boolean)
+        .join("\n");
 
+      const diagnostic = new vscode.Diagnostic(
+        range,
+        message,
+        getSeverityFromCategory(incident.violation_category),
+      );
       diagnostic.source = "konveyor";
-      diagnostic.code = code;
+      diagnostic.code = `${incident.violationId}-${incident.uri}-${incident.lineNumber}`;
```

Comment:
1. Traditionally the field is used to identify the type of problem i.e. code from the compiler to point to compilation error. Usually they are links that open some doc page. Converting it into UID may confuse other tooling. 
3.  `uri` and `lineNumber` can be retrieved from diagnostics (i.e. from Range)

Reply:
https://vscode-api.js.org/classes/vscode.Diagnostic.html#code

> A code or identifier for this diagnostic. Should be used for later processing, e.g. when providing [code actions](https://vscode-api.js.org/interfaces/vscode.CodeActionContext.html).